src/flamenco/runtime/fd_blockstore.h

```diff
@@ -410,6 +429,20 @@ struct fd_blockstore_archiver {
 typedef struct fd_blockstore_archiver fd_blockstore_archiver_t;
 #define FD_BLOCKSTORE_ARCHIVE_START sizeof(fd_blockstore_archiver_t)
 
+/*   CONCURRENCY NOTES FOR BLOCKSTORE ENJOINERS:
+
+   With the parallelization of the shred map and block map, parts of the
+   blockstore are concurrent, and parts are not. Block map and shred map
+   have their own locks, which are managed through the
+   query_try/query_test APIs. When accessing buf_shred_t and
+   block_meta_t items then, the caller does not need to use
+   blockstore_start/end_read/write. However, the
+   blockstore_start/end_read/write still protects the blockstore_shmem_t
+   object. If you are reading and writing any blockstore_shmem fields
+   and at the same time accessing the block_meta_t or buf_shred_t, you
+   should call both the blockstore_start/end_read/write APIs AND the map
+   query_try/test APIs. These are locks of separate concerns and will
+   not deadlock with each other. */
```

Comment:
we'll clean this up to be only fenced reads / writes of primitive fields on shmem_t